### PR TITLE
Kbucket refactoring for Content Routing

### PR DESCRIPTION
**Pull request by aarshkshah1992**
@Stebalien 

1. Export the `Option` type so we can refer to in in the DHT.
2. Fix interface conversion in RT.Print().
3. Functions to get the number of peers for a Cpl/determine if a bucket is full.

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ type options struct {
 }
 
 // Apply applies the given options to this option.
-func (o *options) Apply(opts ...option) error {
+func (o *options) Apply(opts ...Option) error {
```
Comment (Stebalien) [inline options.go]: note: `Apply` can be private.

File: table.go [hunk 2]
```diff
@@ -196,6 +196,34 @@ func (rt *RoutingTable) peersToValidate() []PeerInfo {
...
+func (rt *RoutingTable) NPeersForCpl(cpl uint) int {
+	rt.tabLock.RLock()
+	defer rt.tabLock.RUnlock()
+
+	// it's in the last bucket
+	if int(cpl) >= len(rt.buckets)-1 {
+		count := 0
+		b := rt.buckets[len(rt.buckets)-1]
+		for _, p := range b.peerIds() {
+			if CommonPrefixLen(rt.local, ConvertPeerID(p)) == int(cpl) {
```
Comment (Stebalien) [inline table.go]: Ouch. We should probably start storing these converted IDs. Sha256 is going to kill us.

File: table.go [hunk 3]
```diff
@@ -459,7 +487,7 @@ func (rt *RoutingTable) Print() {
 		fmt.Printf("\tbucket: %d\n", i)
 
 		for e := b.list.Front(); e != nil; e = e.Next() {
-			p := e.Value.(peer.ID)
+			p := e.Value.(*PeerInfo).Id
```
Comment (Stebalien) [inline table.go]: Let's call this in a test.

File: table.go (hunk 2 above)
Comment (aarshkshah1992) [reply table.go]: @Stebalien Great point. I never thought about the cost we pay every time we call this function. I've created an issue here:

https://github.com/libp2p/go-libp2p/issues/812

**Pull request merged**